include/GafferSceneUI/StandardLightVisualiser.h

```diff
@@ -0,0 +1,86 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2015, John Haddon. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef GAFFERSCENEUI_STANDARDLIGHTVISUALISER_H
+#define GAFFERSCENEUI_STANDARDLIGHTVISUALISER_H
+
+#include "GafferSceneUI/LightVisualiser.h"
+#include "IECore/Light.h"
+#include "IECore/SimpleTypedData.h"
+#include "IECoreGL/Group.h"
+
+
+namespace GafferSceneUI
+{
+
+/// Class for performing standard visualisations of lights,
+/// by mapping shader parameters to features of the visualisation.
+/// This also provides several protected utility methods for
+/// making standard visualisations, so is suitable for use as
+/// a base class for custom light visualisers.
+class StandardLightVisualiser : public LightVisualiser
+{
+
+	public :
+
+		IE_CORE_DECLAREMEMBERPTR( StandardLightVisualiser )
+
+		StandardLightVisualiser();
+		virtual ~StandardLightVisualiser();
+
+		virtual IECoreGL::ConstRenderablePtr visualise( const IECore::ObjectVector *shaderVector, IECoreGL::ConstStatePtr &state ) const;
+
+	protected :
+		static void addEnvLightVisualiser( IECoreGL::GroupPtr &output, Imath::Color3f multiplier, const std::string &textureName );
```

Comment:
To me, these still don't seem in keeping with the rest of the API (spotlightCone() etc) :

- The rest return things, these push into a state or group.
- spotlightCone() takes an inner and outer angle to avoid the ambiguity of penumbraType, but basicLightVisualiser() takes this penumbraType thing from the metadata.
- Other small issues
  - I presume multiplier should be color to match colorIndicator?
  - If a function doesn't maintain a reference to an argument after calling, we should pass `Group *` rather than `GroupPtr`

Unless you need them public for some reason, I suggest we just hide them for now, or if you have other stuff to move onto I can make some tweaks after we merge.